stakingModel/test/GoodFundManager.e2e.test.js

```diff
@@ -0,0 +1,117 @@
+    avatarAddress = dao_addresses.Avatar;
+    ubiBridgeRecipient = staking_addresses.UBIScheme;
+    dai = await DAIMock.at(staking_addresses.DAI);
+    cDAI = await cDAIMock.at(staking_addresses.cDAI);
+    simpleStaking = await SimpleDAIStaking.at(staking_addresses.DAIStaking);
+    goodReserve = await GoodReserve.at(staking_addresses.Reserve);
+    goodFundManager = await GoodFundsManager.at(staking_addresses.FundManager);
+    marketMaker = await MarketMaker.at(staking_addresses.MarketMaker);
+    contribution = await ContributionCalculation.at(staking_addresses.Contribution);
+    goodDollar = await GoodDollar.at(dao_addresses.GoodDollar);
+    registrar = await SchemeRegistrar.at(dao_addresses.SchemeRegistrar);
+    absoluteVote = await AbsoluteVote.at(dao_addresses.AbsoluteVote);
+    deploy_settings = await fse.readFile("../migrations/deploy-settings.json", "utf8");
+    console.log(deploy_settings);
+    // schemes
+    setReserve = await FundManagerSetReserve.new(avatarAddress, goodFundManager.address, goodReserve.address);
+  });
+
+  it("should mints cdai and updates exchange rate", async () => {
+    await dai.mint(staker, web3.utils.toWei("100", "ether"));
+    await dai.approve(simpleStaking.address, web3.utils.toWei("100", "ether"), {
+      from: staker
+    });
+    await simpleStaking
+    .stakeDAI(web3.utils.toWei("100", "ether"), {
+      from: staker
+    })
+    .catch(console.log);
+    await cDAI.exchangeRateCurrent();
+    let rate = await cDAI.exchangeRateStored();
+    expect(rate.toString()).to.be.equal("10201010101010101010101010101");
+  });
+
+  it("should be able to set the reserve", async () => {
+    const executeProposalEventExists = await proposeAndRegister(setReserve.address,
+      registrar, proposalId, absoluteVote, avatarAddress, founder);
+    // Verifies that the ExecuteProposal event has been emitted
+    expect(executeProposalEventExists).to.be.true;
+    await setReserve.setReserve();
+    const reserve1 = await goodFundManager.reserve();
+    expect(reserve1).to.be.equal(goodReserve.address)
+  });
+
+  it("should not mint UBI if not in the interval", async () => {
+    const error = await goodFundManager
+      .transferInterest(simpleStaking.address)
+      .catch(e => e);
+    expect(error.message).to.have.string("wait for the next interval");
+  });
+
+  it("should collect the interest and transfer it to the reserve and recieves minted gd back to the staking contract", async () => {
+    await next_interval();
+    await cDAI.exchangeRateCurrent();
+    let recipientBefore = await goodDollar.balanceOf(ubiBridgeRecipient);
+    await goodFundManager.transferInterest(simpleStaking.address);
+    let recipientAfter = await goodDollar.balanceOf(ubiBridgeRecipient);
+    let stakingGDBalance = await goodDollar.balanceOf(simpleStaking.address);
+    expect(stakingGDBalance.toString()).to.be.equal("0"); //100% of interest is donated, so nothing is returned to staking
```

Comment:
need test for when avgDonatedInterest is not 100%, maybe not e2e, but in unit tests with SimpleStakingMock

Reply:
already exists (80%)